benchmarking/commons/hpo_main_simulator.py

```diff
@@ -112,85 +149,54 @@ def subsample(
+        command line arguments
+    :param post_processing: Called after tuning has finished, passing the tuner
+        as argument. Can be used for postprocessing, such as output or storage
+        of extra information
+    :param map_method_args: See above, optional
+    :param tuning_job_metadata: Metadata added to the tuner, can be used to manage results
+    :param use_transfer_learning: If True, we use transfer tuning. Defaults to
+        False
     """
-    if extra_args is None:
-        extra_args = []
-    else:
-        extra_args = extra_args.copy()
-    nested_dict = is_dict_of_dict(benchmark_definitions)
-    extra_args.extend(
-        [
-            dict(
-                name="benchmark",
-                type=str,
-                help="Benchmark to run from benchmark_definitions",
-            ),
-            dict(
-                name="verbose",
-                type=int,
-                default=0,
-                help="Verbose log output?",
-            ),
-            dict(
-                name="support_checkpointing",
-                type=int,
-                default=1,
-                help="If 0, trials are started from scratch when resumed",
-            ),
-            dict(
-                name="fcnet_ordinal",
-                type=str,
-                choices=("none", "equal", "nn", "nn-log"),
-                default="nn-log",
-                help="Ordinal encoding for fcnet categorical HPs with numeric values. Use 'none' for categorical encoding",
-            ),
-            dict(
-                name="restrict_configurations",
-                type=int,
-                default=0,
-                help="If 1, scheduler only suggests configs contained in tabulated benchmark",
-            ),
-        ]
+    configuration.check_if_all_paremeters_present(
+        LOCAL_LOCAL_SIMULATED_BENCHMARK_REQUIRED_PARAMETERS
     )
-    if nested_dict:
-        extra_args.append(
-            dict(
-                name="benchmark_key",
-                type=str,
-                required=True,
-            )
-        )
-    args, method_names, seeds = _parse_args(methods, extra_args)
-    args.verbose = bool(args.verbose)
-    args.support_checkpointing = bool(args.support_checkpointing)
-    args.restrict_configurations = bool(args.restrict_configurations)
-    if args.benchmark is not None:
-        benchmark_names = [args.benchmark]
+    configuration.expand_base_arguments(
+        LOCAL_LOCAL_SIMULATED_BENCHMARK_REQUIRED_PARAMETERS
+    )
+
+    nested_dict = is_dict_of_dict(benchmark_definitions)
+    if configuration.benchmark is not None:
+        benchmark_names = [configuration.benchmark]
     else:
         if nested_dict:
             # If ``parse_args`` is called from ``launch_remote``, ``benchmark_key`` is
             # not set. In this case, ``benchmark_names`` is not needed
-            k = args.benchmark_key
+            k = configuration.benchmark_key
```

Comment:
You have not entered this above into `LOCAL_LOCAL_SIMULATED_BENCHMARK_REQUIRED_PARAMETERS`, so it cannot be present here.

Reply:
This is important. The use case of also parallelizing over benchmarks is something we would like to have working properly.